app/javascript/components/ansible-repository-form/helpers.jsx

```diff
@@ -0,0 +1,21 @@
+const GitUrlParse = require('git-url-parse');
+
+export const urlValidation = (values) => {
+  const errors = {};
+  try {
+    const parsed = values.scm_url ? GitUrlParse(values.scm_url) : {};
+    if ((parsed.protocol !== 'http'
+          && parsed.protocol !== 'https'
+          && parsed.protocol !== 'file'
+          && parsed.protocol !== 'ssh')
+          || parsed.pathname === '') {
+      errors.scm_url = __('URL must include a protocol (http://, https:// or file://) with path or be a valid SSH path (user@server:path or ssh://user@address:port/path)');
+    }
+    if (parsed.protocols.length > 2 || (parsed.protocols.length === 1 &&  parsed.protocols[0] !== parsed.protocol)) {
+      errors.scm_url = __('URL must include a protocol (http://, https:// or file://) with path or be a valid SSH path (user@server:path or ssh://user@address:port/path)');
+    }
+  } catch (err) {
+    errors.scm_url = __('URL must include a protocol (http://, https:// or file://) with path or be a valid SSH path (user@server:path or ssh://user@address:port/path)');
+  }
```

Comment:
Think we should extract the strings from the logic and do it only once.
```suggestion
    var error = false
    const parsed = values.scm_url ? GitUrlParse(values.scm_url) : {};
    if ((parsed.protocol !== 'http'
          && parsed.protocol !== 'https'
          && parsed.protocol !== 'file'
          && parsed.protocol !== 'ssh')
          || parsed.pathname === '') {
      error = true
    }
    if (parsed.protocols.length > 2 || (parsed.protocols.length === 1 &&  parsed.protocols[0] !== parsed.protocol)) {
      error = true
    }
  } catch (err) {
    error = true
  }
  
  if (error) {
    errors.scm_url = __('URL must include a protocol (http://, https:// or file://) with path or be a valid SSH path (user@server:path or ssh://user@address:port/path)');
  }
```